pix2pix/rest.py

```diff
@@ -0,0 +1,62 @@
+#!usr/bin/env python
+#
+# Copyright (c) 2019 Intel Corporation
+#
+# Main author:
+#   * unrahul <rahul.unnikrishnan.nair@intelcom>
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+"""a rest api for pix2pix."""
+import io
+
+import flask
+from PIL import Image
+
+from infer import infer
+
+app = flask.Flask("pix2pix api")
+
+
+banner = {
+    "what": "pix2pix api",
+    "usage": {
+        "client": "curl -i  -X POST -F img=@data/abstract.jpg 'http://localhost:5000/generate'",
+        "server": "docker run -d -p 5000:5000 pix2pix-infer",
+    },
+}
+
+
+@app.route("/usage", methods=["GET"])
+def usage():
+    return flask.jsonify({"info": banner}), 201
+
+
+@app.route("/generate", methods=["POST"])
+def generate():
+    """generate an image from an `img` obtained from a post request"""
+    if flask.request.files.get("img"):
+        img = flask.request.files["img"].read()
+        img = Image.open(io.BytesIO(img))
+        y_hat = infer(img)
```

Comment:
Doesn't infer take a path? not an image file?

Reply:
yup!, need to change that